File: java/src/main/java/bparolini/dsa/neetcode/ContainsDuplicate.java
package bparolini.dsa.neetcode;

import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> item = new HashSet<Integer>();

        for (int num : nums) {
            if (!item.add(num)) {
                return true;
            }
        }

        return false;
    }
}

File: go/leetcode/contains_duplicate.go
package leetcode

import "slices"

func containsDuplicate(nums []int) bool {
	item := make([]int, len(nums))

	for _, num := range nums {
		if slices.Contains(item, num) {
			return true
		}

		item = append(item, num)
	}

	return false
}
